### ISSUE TITLE
Xbox Fix Host Save NonFunctional

### ISSUE THREAD

**Issue by Sat727** [deafdudecomputers/PalWorldSaveTools#18]
I am having an issue with logging into the server after using the fix host save function. I've followed the instructions within the readme file.

Error after successfully using the "fix_host_save" function within the project. The script runs with the following stdout which is seemingly successful:

```Enter new GUID: F1A4F926000000000000000000000000
Enter old GUID: 00000000000000000000000000000001
WARNING: Running this script WILL change your save files... Backup recommended.
Press enter to continue...
Converting LocalWorldSave\Level.sav to JSON
Decompressing sav file
Loading GVAS file
Converting LocalWorldSave\Players\00000000000000000000000000000001.sav to JSON
Decompressing sav file
Loading GVAS file
Modifying JSON save data...
Converting data to SAV, saving to LocalWorldSave\Level.sav
Compressing SAV file
Compressing: 99% completedWriting SAV file to LocalWorldSave\Level.sav
Converting data to SAV, saving to LocalWorldSave\Players\00000000000000000000000000000001.sav
Compressing SAV file
Writing SAV file to LocalWorldSave\Players\00000000000000000000000000000001.sav
Fix has been applied! Have fun!
```
However, when logging into the server I get a joined message, and a disconnected message immediately after by my account, on the game I get sent to the main menu with the error message:  

``"Failed Join Game. [LocalPlayerId:256][6F89120900...] Result = FailedInvalidLoginPlayerCharacterHandle" ``

I'm guessing this does not support Xbox or Microsoft version yet as it seemingly works for my Steam version. If needed, I can send over my sav files.

There may be a difference between the Xbox/Microsoft and Steam level.sav or player files causing this error. 

This used to work with the palworld-save-tools repo but it is abandoned by it's creator.

I appreciate this project and this deserves more attention.

**Comment by Sat727** [comment]
TLDR; I was able to fix this; here's how

using XPG save extractor created, the 01.sav file in the Level folder. Use the convert.py from this tool to convert 01.sav to JSON. Then re-convert it back to a .sav file. Rename the new 01.sav Level.sav and profit

Ultimately this must be something weird with XPG save extractor and not with this tool at all; Kudos to the creator

**Comment by ydbdyds** [comment]
I'm having the same problem, but server-to-server.
